web/src/settings_users.ts

```diff
@@ -191,29 +191,56 @@ function role_selected_handler(
     widget.render();
 }
 
-function get_roles(): {
-    unique_id: number;
-    name: string;
-}[] {
+function count_users_by_role(user_ids: number[]): Record<number, number> {
+    const role_counts: Record<number, number> = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        role_counts[role_code] = (role_counts[role_code] ?? 0) + 1;
+    }
+
+    return role_counts;
+}
+
+function get_roles_with_counts(user_ids: number[]): {unique_id: number; name: string}[] {
+    const role_counts = count_users_by_role(user_ids);
     return [
-        {unique_id: 0, name: $t({defaultMessage: "All roles"})},
+        {
+            unique_id: 0,
+            name: $t(
+                {defaultMessage: "{all_roles} ({count})"},
+                {
+                    all_roles: $t({defaultMessage: "All roles"}),
```

Comment:
I found that all roles was not being translated when changing the language. After implementing this solution, they are now translating correctly when the language is changed.

Reply:
It was not translated because there are no translations for "All roles ({count})" text. See https://zulip.readthedocs.io/en/latest/translating/index.html for how translations work.

Reply:
> It was not translated because there are no translations for "All roles ({count})" text. See https://zulip.readthedocs.io/en/latest/translating/index.html for how translations work.

Yes, that's correct. I thought that if we use that method, we would need to add new translations. By doing it this way, we can save translation work.

However, if you prefer, I can revert the changes to the previous version.

Reply:
> Yes, that's correct. I thought that if we use that method, we would need to add new translations. By doing it this way, we can save translation work.
> 
> However, if you prefer, I can revert the changes to the previous version.

@sahil839 , is this approach correct?

Reply:
> > It was not translated because there are no translations for "All roles ({count})" text. See https://zulip.readthedocs.io/en/latest/translating/index.html for how translations work.
> 
> Yes, that's correct. I thought that if we use that method, we would need to add new translations. By doing it this way, we can save translation work.
> 
> However, if you prefer, I can revert the changes to the previous version.

@sahil839 , Please give your thoughts over this.
Thanks!

Reply:
You should revert to the previous version here. That is the usual workflow we follow where translations for different languages are added by translators.